src/Microsoft.OData.Core/JsonLight/ODataJsonLightReader.cs

```diff
@@ -2413,8 +2483,6 @@ private void StartNestedPropertyInfo(ODataJsonLightReaderNestedPropertyInfo read
         private void StartNestedStreamInfo(ODataJsonLightReaderStreamInfo readerStreamInfo)
         {
             Debug.Assert(readerStreamInfo != null, "readerNestedResourceInfo != null");
-            Debug.Assert(this.jsonLightResourceDeserializer.JsonReader.CanStream() || this.CurrentScope is JsonLightResourceSetScope,

```

Comment:
Explanation for this change provided [here](https://github.com/OData/odata.net/pull/2335/files#r826681344)